mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -129,17 +129,21 @@ object Bucketizer extends DefaultParamsReadable[Bucketizer] {
     if (feature == splits.last) {
       splits.length - 2
     } else {
-      val idx = ju.Arrays.binarySearch(splits, feature)
-      if (idx >= 0) {
-        idx
+      if (feature.isNaN) {
```

Comment:
I think this might just be me, but, it felt like the first two checks were at the same level:

```
if (...) {
} else if (...) {
} else {
  ...
}
```

Or alternatively, I could see handling the `feature == splits.last` case in the `idx >= 0` case. Really that's the case that handle exact match of a value and right now it's split across two clauses. Then it makes sense to start the method by handling NaN alone.